data-model/src/main/java/io/micronaut/data/model/PersistentEntity.java

```diff
@@ -405,6 +405,14 @@ default PersistentPropertyPath getPropertyPath(@NonNull String[] propertyPath) {
     @NonNull
     Optional<NamingStrategy> findNamingStrategy();
 
+    /**
+     * Obtains a PersistentProperty representing id or version property by name.
+     *
+     * @param name The name of the id or version property
+     * @return The PersistentProperty used as id or version or null if it doesn't exist
+     */
+    PersistentProperty getIdOrVersionPropertyByName(String name);
```

Comment:
Add `@since`

Reply:
This existed in `SourcePeristentEntity` so I just added it here and implemented in `RuntimePersistentEntity`

Reply:
Donr